WordPress/src/main/java/org/wordpress/android/ui/reader/repository/ReaderPostRepository.kt

```diff
@@ -3,451 +3,182 @@ package org.wordpress.android.ui.reader.repository
+                is OnPostLikeUnChanged -> { }
+            }
+        }
+    }
+
+    private fun onNewPosts(event: UpdatePostsEnded) {
+        reloadPosts()
+    }
+
+    private fun onChangedPosts(event: UpdatePostsEnded) {
+        reloadPosts()
+    }
+
+    private fun onUnchanged(event: UpdatePostsEnded) {
+    }
+
+    private fun onFailed(event: UpdatePostsEnded) {
+        _communicationChannel.postValue(
+                Event(ReaderRepositoryCommunication.Error.RemoteRequestFailure))
+    }
 
-    private suspend fun getMockDiscoverFeed(): LiveData<ReaderPostList> {
-        return withContext(ioDispatcher) {
-            mutableDiscoveryFeed.postValue(ReaderPostList.fromJson(JSONObject(discoverJson)))
-            discoveryFeed
+    private fun onActivePosts() {
+        loadPosts()
+    }
+
+    private fun onInactivePosts() {
+    }
+
+    private fun loadPosts() {
+        launch {
+            val existsInMemory = posts.value?.let {
+                !it.isEmpty()
+            } ?: false
+            val refresh = shouldAutoUpdateTagUseCase.get(readerTag)
+
+            if (!existsInMemory) {
+                val result = getPostsForTagUseCase.get(readerTag)
+                _mutablePosts.postValue(result)
+            }
+
+            if (refresh) {
+                val response = fetchPostsForTagUseCase.fetch(readerTag)
+                if (response != Success) _communicationChannel.postValue(Event(response))
+            }
+        }
+    }
+
+    private fun reloadPosts() {
+        launch {
+            val result = getPostsForTagUseCase.get(readerTag)
+            _mutablePosts.postValue(result)
         }
     }
+
+    class Factory
+    @Inject constructor(
+        @Named(BG_THREAD) private val bgDispatcher: CoroutineDispatcher,
+        private val getPostsForTagUseCase: GetPostsForTagUseCase,
+        private val getNumPostsForTagUseCase: GetNumPostsForTagUseCase,
+        private val shouldAutoUpdateTagUseCase: ShouldAutoUpdateTagUseCase,
+        private val getPostsForTagWithCountUseCase: GetPostsForTagWithCountUseCase,
+        private val fetchPostsForTagUseCase: FetchPostsForTagUseCase,
+        private val readerUpdatePostsEndedHandler: ReaderUpdatePostsEndedHandler,
+        private val postLikeActionUseCase: PostLikeActionUseCase
+    ) {
+        fun create(readerTag: ReaderTag): ReaderPostRepository {
+            return ReaderPostRepository(
+                    bgDispatcher,
+                    readerTag,
+                    getPostsForTagUseCase,
+                    getNumPostsForTagUseCase,
+                    shouldAutoUpdateTagUseCase,
+                    getPostsForTagWithCountUseCase,
+                    fetchPostsForTagUseCase,
+                    readerUpdatePostsEndedHandler,
+                    postLikeActionUseCase
+
+            )
+        }
+    }
+}
+
+// todo: annmarie once it's decided to keep this, then move out of repository file
+sealed class OnReaderRepositoryEvent {
```

Comment:
New Event declarations. This can be built out to support other reader actions (bookmark), etc.